internal/application/application.go

```diff
@@ -136,16 +140,37 @@ func Exists(ctx context.Context, cluster *kubernetes.Cluster, app models.AppRef)
 	return true, nil
 }
 
-// CurrentlyStaging returns true if there is an active Job for this application.
-func CurrentlyStaging(ctx context.Context, cluster *kubernetes.Cluster, namespace, appName string) (bool, error) {
+// IsCurrentlyStaging returns true if the app is staging (there is an active Job for this application).
+// If you need this information for more than one app please use the CurrentlyStaging(ctx context.Context, cluster JobLister, namespace string)
+func IsCurrentlyStaging(ctx context.Context, cluster JobLister, namespace, appName string) (bool, error) {
+	currentlyStagingApps, err := currentlyStaging(ctx, cluster, namespace, appName)
+	return currentlyStagingApps[appName], err
+}
+
+// CurrentlyStaging returns a map of applications that are currently in a staging status
+func CurrentlyStaging(ctx context.Context, cluster JobLister, namespace string) (map[string]bool, error) {
+	return currentlyStaging(ctx, cluster, namespace, "")
+}
 
-	// Check all jobs for the app for activity.
-	selector := fmt.Sprintf("app.kubernetes.io/name=%s,app.kubernetes.io/part-of=%s",
-		appName, namespace)
+// currentlyStaging is a utility func that will load a map of the status of the application's staging jobs
+// If no appName is specified it will load a complete map, otherwise the map will contain just the status of job of the specified app
+func currentlyStaging(ctx context.Context, cluster JobLister, namespace, appName string) (map[string]bool, error) {
+	stagingJobsMap := make(map[string]bool)
+
+	// filter the jobs in the namespace
```

Comment:
Does this work for `epinio app list --all` ? This listing across all namespaces is signaled by `namespace == ""`.

Reply:
The previous implementation was using the same logic with the plain labels (not in a map, but I guess it's the same).
```
selector := fmt.Sprintf("app.kubernetes.io/name=%s,app.kubernetes.io/part-of=%s", appName, namespace)
```

 I'll do a manual check to be sure.

We should have some tests for the `--all` flag, anyway. If not, we should add them. :slightly_smiling_face:
I'll have a look also at this. :+1:

Reply:
__Beware__ The list shows the status as `1/1` etc. and that info comes out of the main app pods.
Not from the staging jobs.
What does the `list --all` show while the apps are actually in staging, not running ?

Reply:
You will need multiple terminals, one per app to put into staging, plus the terminal where `list --all` is run.
Best to find some app which takes long(er) to stage.

Reply:
In the end there was an error about namespaces, so thanks for blocking this!

I'll probably open one more PR with some tests to have a wider coverage of this APIs. :+1: